Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -14418,8 +14411,38 @@ def get_parsed_incident_entries(incident_entries_dict: dict[str, list[dict[str,
     return parsed_incident_entries
 
 
+def update_offset_dict(incident_entries_dict: dict[str, list[dict[str, Any]]],
+                       last_fetch_dict: LastFetchTimes,
+                       offset_fetch_dict: Offset):
+    """
+        for each log type incident entries array, parse the raw incidents into context incidents.
+        if necessary, update the latest fetch time and last ID values in their corresponding dictionaries.
+
+        Args:
+            incident_entries_dict (Dict[str, List[Dict[str, Any]]]): list of dictionaries representing raw incident entries
+            last_fetch_dict (Dict[str, str]): last fetch dictionary
+            offset_fetch_dict (Dict[str, Dict]): last offset dictionary
+
+    """
+    for log_type, incident_entries in incident_entries_dict.items():
+        if incident_entries:
+            last_fetch_time = dateparser.parse(last_fetch_dict.get(log_type, ''), settings={'TIMEZONE': 'UTC'})  # type: ignore[literal-required]
+            for entry in incident_entries:
+                time_field = 'match_time' if log_type == 'Correlation' else 'time_generated'
+                log_time = dateparser.parse(entry.get(time_field, ''), settings={'TIMEZONE': 'UTC'})
+
+                if (not log_type in offset_fetch_dict or
+                    not last_fetch_time or
+                    log_time > last_fetch_time):
+                    offset_fetch_dict[log_type] = 1  # type: ignore[literal-required]
+                    last_fetch_time = log_time
+                elif log_time == last_fetch_time:
+                    offset_fetch_dict[log_type] += 1  # type: ignore[literal-required]
+
```

Comment:
```suggestion
                else:
                    break
```

Reply:
We assume that the incidents are ordered no? In that case if wont happen